zerver/lib/users.py

```diff
@@ -606,12 +606,16 @@ def get_raw_user_data(
     for row in user_dicts:
         if profiles_by_user_id is not None:
             custom_profile_field_data = profiles_by_user_id.get(row["id"], {})
+        client_gravatar_for_user = (
+            client_gravatar
+            and row["email_address_visibility"] == UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE
```

Comment:
Would it be better for this check to use `can_access_delivery_email(acting_user, ...)`?